application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadLifecycleListener.java

```diff
@@ -0,0 +1,134 @@
+package org.togetherjava.tjbot.features.help;
+
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.entities.channel.forums.ForumTag;
+import net.dv8tion.jda.api.events.channel.update.ChannelUpdateAppliedTagsEvent;
+import net.dv8tion.jda.api.events.channel.update.ChannelUpdateArchivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.features.EventReceiver;
+
+import java.time.Instant;
+import java.util.List;
+
+import static org.togetherjava.tjbot.db.generated.tables.HelpThreads.HELP_THREADS;
+
+/**
+ * Listens for help thread events after creation of thread. Updates metadata based on those events
+ * in database.
+ */
+public final class HelpThreadLifecycleListener extends ListenerAdapter implements EventReceiver {
+    private static final Logger logger = LoggerFactory.getLogger(HelpThreadLifecycleListener.class);
+    private final HelpSystemHelper helper;
+    private final Database database;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param helper to work with the help threads
+     * @param database the database to store help thread metadata in
+     */
+    public HelpThreadLifecycleListener(HelpSystemHelper helper, Database database) {
+        this.helper = helper;
+        this.database = database;
+    }
+
+    @Override
+    public void onChannelUpdateArchived(@NotNull ChannelUpdateArchivedEvent event) {
+        ThreadChannel threadChannel = event.getChannel().asThreadChannel();
+
+        if (!helper.isHelpForumName(threadChannel.getParentChannel().getName())) {
+            return;
+        }
+        handleThreadStatus(threadChannel);
+    }
+
+    @Override
+    public void onChannelUpdateAppliedTags(@NotNull ChannelUpdateAppliedTagsEvent event) {
+        ThreadChannel threadChannel = event.getChannel().asThreadChannel();
+
+        if (!helper.isHelpForumName(threadChannel.getParentChannel().getName())
+                || shouldIgnoreUpdatedTagEvent(event)) {
+            return;
+        }
+
+
+        List<String> newlyAppliedTagsOnly = event.getAddedTags()
+            .stream()
+            .filter(helper::shouldIgnoreTag)
+            .map(ForumTag::getName)
+            .toList();
+
+        String tags = String.join(", ", newlyAppliedTagsOnly);
```

Comment:
- `getAddedTags` will not account for removed tags so change it to `getNewTags`
- use collector
- remove space after comma

Reply:
> `getAddedTags` will not account for removed tags so change it to getNewTags

wym, i don't wanna keep track of false tags. If category is updated by someone else, i always want the latest applied tag. For example, let's say someone starts with Java, Frameworks tag due to poor judgement on their part and it's later changed to Other category. Then it doesn't make sense to keep track of `Java, Frameworks, Other` instead i only want `Other` in there.

Reply:
from the [docs](https://javadoc.io/doc/net.dv8tion/JDA/latest/net/dv8tion/jda/api/events/channel/update/ChannelUpdateAppliedTagsEvent.html), if someone starts with `Java,Frameworks` and then remove `Frameworks` `getAddedTags` will be empty, and `getNewTags` will have `Java` which is what we want to track

if someone starts with `Java,Frameworks` and then add `Other` while removing `Java,Frameworks` `getAddedTags` will have `Others` and `getNewTags` will have `Others` as well